### PR TITLE
[iOS] Add turnstile events for Snapshotter

### PR DESCRIPTION
**Pull request by lloydsheng**
Snapshotter (i.e. https://www.mapbox.com/android-docs/maps/overview/snapshotter) is part of the Maps SDK and should be sending turnstile events before usage on both iOS and Android (Snapshotter requests need the same access token we use for regular tile requests).
We should add turnstile events for Snapshotter.

### CODE REVIEW

File: platform/darwin/src/MGLMapSnapshotter.mm [hunk 1]
```diff
@@ -74,6 +75,7 @@ - (instancetype)initWithImage:(nullable MGLImage *)image scale:(CGFloat)scale po
         _latLngForFn = std::move(latLngForFn);
         _scale = scale;
         _image = image;
+        [MGLMapboxEvents pushTurnstileEvent];
```
Comment (julianrex) [inline platform/darwin/src/MGLMapSnapshotter.mm]: I think this should be in `-startWithQueue:completionHandler:` - an `MGLMapSnapshotter` may be allocated and not used, or used multiple times.

Perhaps sending the turnstile event somewhere around 

https://github.com/mapbox/mapbox-gl-native/blob/53351c2ae83618eb8324f9ed688fdbf09debf21a/platform/darwin/src/MGLMapSnapshotter.mm#L201-L210

would make sense?

Comment (julianrex) [reply platform/darwin/src/MGLMapSnapshotter.mm]: It looks like Android sends it in the initializer too: https://github.com/mapbox/mapbox-gl-native/pull/13475

Which should it be? /cc @tobrun

Comment (lloydsheng) [reply platform/darwin/src/MGLMapSnapshotter.mm]: @julianrex  I think the behavior should consist with `MGLMapview` did.

https://github.com/mapbox/mapbox-gl-native/blob/8bc362e3a2836cfa96ef14a41a0e882a1d08e881/platform/ios/src/MGLMapView.mm#L629-L632

The turnstile events send in the initializer of mapview.

Comment (lloydsheng) [reply platform/darwin/src/MGLMapSnapshotter.mm]: @julianrex @tobrun What's your opinion? Can I merge this PR?

**Comment by springmeyer** [comment]
> We can go ahead and merge this for consistency (with the existing landed Android PR) - though I'd like to fully understand why it shouldn't be in the start... method.

Is there a ticket tracking this question? We should get to a resolution. /cc @lloydsheng @julianrex

**Pull request merged**